components/gitpod-protocol/src/gitpod-service.ts

```diff
@@ -353,16 +353,24 @@ export interface ClientHeaderFields {
     clientRegion?: string;
 }
 
-export const WORKSPACE_TIMEOUT_DEFAULT_SHORT = "short";
-export const WORKSPACE_TIMEOUT_DEFAULT_LONG = "long";
-export const WORKSPACE_TIMEOUT_EXTENDED = "extended";
-export const WORKSPACE_TIMEOUT_EXTENDED_ALT = "180m"; // for backwards compatibility since the IDE uses this
-export const WorkspaceTimeoutValues = [
-    WORKSPACE_TIMEOUT_DEFAULT_SHORT,
-    WORKSPACE_TIMEOUT_DEFAULT_LONG,
-    WORKSPACE_TIMEOUT_EXTENDED,
-    WORKSPACE_TIMEOUT_EXTENDED_ALT,
-] as const;
+export type WorkspaceTimeoutDuration = string;
+export namespace WorkspaceTimeoutDuration {
+    export function validate(duration: string): WorkspaceTimeoutDuration {
+        const unit = duration.slice(-1);
+        if (!["m", "h", "d"].includes(unit)) {
+            throw new Error(`Invalid timeout unit: ${unit}`);
+        }
+        const value = parseInt(duration.slice(0, -1));
+        if (isNaN(value) || value <= 0) {
+            throw new Error(`Invalid timeout value: ${duration}`);
+        }
+        return duration;
+    }
+}
```

Comment:
Would it be easier to represent this as Seconds? 

The constants can then be expressed as WorkspaceTimeDuration * 60 * 30

The CLI can can parse 30s, but send it as a # of seconds

Reply:
Yes, `workspaceTimoutSeconds` would be clearer. I wanted to avoid breaking API and decided to go with something that is compatible to the `"180m"` string that is currently passed.